MeasureAuthoringTool/mat-drupal
# Docker deployment updates

## Description
Updates the Dockerfile and related settings to make deployment to docker easier.

## To Test Locally
1. Comment out the PDO line in `mat.settings.php`.
2. Copy `.env.example` to `.env` and provide some values for the MySQL variables. Doesn't really matter what you use, except that `MYSQL_HOSTNAME` should match docker-compose service name for MySQL ("database").
3. Start docker-compose `docker-compose up -d --build`.
4. Install drupal `docker-compose exec drupal drush site-install minimal --site-name "Measure Authoring Tool" --existing-config -y`.
5. Clear the cache `docker-compose exec drupal drush cr`.
6. Visit http://localhost:8080 to view the site.
7. To login, if the above install command provided an admin username/password, you can use that at http://localhost:8080/user/login. Otherwise:

### Post-Install Login instructions
1. Run `docker-compose exec drupal drush uli`. That command will generate a URL like `http://default/user/reset/1/1581541762/VGGpYbY3vxoFuzAIuXII7K2csKjpqCN9tU4Ed-fWos8/login`.
2. Copy everything after `default` and past that after the site url, eg `http://localhost:8080/user/reset/1/1581541762/VGGpYbY3vxoFuzAIuXII7K2csKjpqCN9tU4Ed-fWos8/login`
